src/main/java/router/RouteProcessor.java

```diff
@@ -41,6 +41,6 @@ private HTTPRequest buildInvalidRequest(ResourceManagementException e) {
 
     private RouteAction selectRouteAction(List<RouteAction> routeActions, HTTPRequest request) {
         Optional<RouteAction> action = routeActions.stream().filter(routeAction -> routeAction.isAppropriate(request)).findFirst();
-        return action.isPresent() ? action.get() : new MethodNotAllowedAction(configuration);
+        return action.isPresent() ? action.get() : new MethodNotAllowedAction(routesFactory.entityHeaderBuilder(), configuration);
```

Comment:
Not happy with this line as I'm reaching into the routes factory to get the EntityHeaderBuilder in order to construct a MethodNotAllowedAction outside of the normal 'flow' e.g. in the RoutesFactory